### PR TITLE
FIX: fails loud if default setting is not set

### PR DESCRIPTION
**Pull request by fantasticfears**
Noted:
- `push_api_secret_key` is set in initializer. Shimed with ''
- `default_theme_key` is set in seeding. Shimed with ''

### CODE REVIEW

File: spec/integrity/site_setting_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  true
+end
+
+describe "site setting integrity checks" do
+  let(:site_setting_file) { File.join(Rails.root, 'config', 'site_settings.yml') }
+  let(:yaml) { YAML.load_file(site_setting_file) }
+
+  it "does not contain invalid interpolation keys" do
+    matches = File.read(site_setting_file).scan(/%\{([^a-zA-Z\s]+)\}|\{\{([^a-zA-Z\s]+)\}\}/)
```
Comment (tgxworld) [inline spec/integrity/site_setting_spec.rb]: Hmm mayeb try to use `I18nInterpolationKeysFinder.find` here or `I18n::INTERPOLATION_PATTERN`

File: spec/integrity/site_setting_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+require "i18n/duplicate_key_finder"
+
+def extract_locale(path)
+  path[/\.([^.]{2,})\.yml$/, 1]
+end
+
+PLURALIZATION_KEYS ||= ['zero', 'one', 'two', 'few', 'many', 'other']
```
Comment (tgxworld) [inline spec/integrity/site_setting_spec.rb]: I believe the I18n gem should already have these in a constant

File: spec/integrity/site_setting_spec.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          expect(setting['default']).not_to be_nil
+        else
+          expect(setting).not_to be_nil
+        end
+
+      end
+    end
+  end
+
+  %w(hidden setting).each do |property|
```
Comment (tgxworld) [inline spec/integrity/site_setting_spec.rb]: What is the `setting` property?

File: spec/integrity/site_setting_spec.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          next unless setting.is_a?(Hash)
+          expect(setting[property] == nil || setting[property] == true).to be_truthy
+        end
+      end
+    end
+  end
+
+  module Boolean; end
+  class TrueClass; include Boolean; end
+  class FalseClass; include Boolean; end
```
Comment (tgxworld) [inline spec/integrity/site_setting_spec.rb]: Why do we need to include an empty module into `TrueClass` and `FalseClass`?

**Comment by tgxworld** [comment]
Thank you for working on this @fantasticfears

File: lib/site_settings/yaml_loader.rb [hunk 5]
```diff
@@ -13,11 +13,13 @@ def load
           # Get default value for the site setting:
           value = hash.delete('default')
           if value.is_a?(Hash)
-            raise Discourse::Deprecation, "Site setting per env is no longer supported. Error setting: #{setting_name}"
+            raise Discourse::Deprecation, "The site setting `#{setting_name}` can no longer be set based on Rails environment. See also `config/environments/<env>.rb`."
+          elsif value.nil?
+            raise Discourse::InvalidParameters, "The site setting `#{setting_name}` is missing default value."
```
Comment (fantasticfears) [inline lib/site_settings/yaml_loader.rb]: Done

File: spec/integrity/site_setting_spec.rb (hunk 1 above)
Comment (fantasticfears) [inline spec/integrity/site_setting_spec.rb]: My bad, these are not related with this integrity test, removed

File: spec/integrity/site_setting_spec.rb (hunk 3 above)
Comment (fantasticfears) [inline spec/integrity/site_setting_spec.rb]: should be `client`, resolved.

**Comment by fantasticfears** [comment]
Looks good and could you help to rerun the test on Travis please?

**Comment by fantasticfears** [comment]
I am at my wit's end for the test suite but rebased run passed.

**Comment by SamSaffron** [comment]
@tgxworld  are you good to merge this in? looks ok to me.
